src/libsyncengine/jobs/network/getappversionjob.cpp

```diff
@@ -0,0 +1,121 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "getappversionjob.h"
+#include "utility/jsonparserutility.h"
+
+#include <config.h>
+
+namespace KDC {
+GetAppVersionJob::GetAppVersionJob(const Platform platform, const std::string &appID) : _platform(platform), _appId(appID) {
+    _httpMethod = Poco::Net::HTTPRequest::HTTP_GET;
+}
+
+std::string platformToStr(Platform platform) {
+    switch (platform) {
+        case Platform::MacOS:
+            return platformMacOsKey;
+        case Platform::Windows:
+            return platformWindowsKey;
+        case Platform::LinuxAMD:
+            return platformLinuxAmdKey;
+        case Platform::LinuxARM:
+            return platformLinuxArmKey;
+        default:
+            return "unknown";
+    }
+}
+
+std::string GetAppVersionJob::getSpecificUrl() {
+    std::stringstream ss;
+    ss << "/app-information/kstore-update/" << platformToStr(_platform) << "/com.infomaniak.drive/" << _appId;
+    return ss.str();
+}
+std::string GetAppVersionJob::getContentType(bool &canceled) {
+    canceled = false;
+    return {};
+}
+
+bool GetAppVersionJob::handleError(std::istream &is, const Poco::URI &uri) {
+    // TODO
+    return false;
+}
+
+DistributionChannel GetAppVersionJob::toDistributionChannel(const std::string &val) const {
+    if (val == versionTypeProdKey) {
+        return DistributionChannel::Prod;
+    }
+    if (val == versionTypeNextKey) {
+        return DistributionChannel::Next;
+    }
+    if (val == versionTypeBetaKey) {
+        return DistributionChannel::Beta;
+    }
+    if (val == versionTypeInternalKey) {
+        return DistributionChannel::Internal;
+    }
+    return DistributionChannel::Unknown;
+}
+
+bool GetAppVersionJob::handleResponse(std::istream &is) {
+    if (!AbstractNetworkJob::handleJsonResponse(is)) {
+        return false;
+    }
+
+    const Poco::JSON::Object::Ptr dataObj = JsonParserUtility::extractJsonObject(jsonRes(), dataKey);
+    if (!dataObj) return false;
+
+    const Poco::JSON::Object::Ptr applicationObj = JsonParserUtility::extractJsonObject(dataObj, applicationKey);
+    if (!applicationObj) return false;
+
+    const Poco::JSON::Array::Ptr publishedVersions = JsonParserUtility::extractArrayObject(applicationObj, publishedVersionsKey);
+    if (!publishedVersions) return false;
+
+    for (const auto &versionInfo : *publishedVersions) {
+        const auto &obj = versionInfo.extract<Poco::JSON::Object::Ptr>();
+        std::string versionType;
+        if (!JsonParserUtility::extractValue(obj, typeKey, versionType)) {
+            return false;
+        }
+
+        const DistributionChannel channel = toDistributionChannel(versionType);
+        if (!JsonParserUtility::extractValue(obj, tagKey, _versionInfo[channel].tag)) {
+            return false;
+        }
+        if (!JsonParserUtility::extractValue(obj, changeLogKey, _versionInfo[channel].changeLog)) {
```

Comment:
You could iterate on a vector of key strings.

Reply:
Not sure how. This is the JSON object I receive in the `published_versions`:
`
{
          "tag": "3.6.4",
          "tag_updated_at": "2024-08-14 12:00:00",
          "version_changelog": "Release notes - ...",
          "type": "production",
          "build_version": "20240814",
          "build_min_os_version": "",
          "download_link": "url",
          "data": [
            "[]"
          ]
}
`

I first need to extract the type. I do not see clearly how iterating over the item of the JSON object would simplify the code. Do you have something precise in mind?